// Copyright 2020 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto2";

package feedwire;

import "components/feed/core/proto/v2/wire/xsurface_content.proto";

option optimize_for = LITE_RUNTIME;

message Stream {}
message Group {}
message PrefetchMetadata {
  optional string uri = 1;
  optional string title = 2;
  optional string image_url = 3;
  repeated string additional_image_urls = 4;
  optional string publisher = 5;
  optional string favicon_url = 6;
  optional string snippet = 7;
  optional string badge_id = 8;
}
message Content {
  enum Alignment {
    ALIGNMENT_UNSPECIFIED = 0;
    DEFAULT = 1;
    CENTER = 2;
  }
  repeated PrefetchMetadata prefetch_metadata = 4;
  optional Alignment alignment = 5;
  optional XSurfaceContent xsurface_content = 1000;
}
